clients/da-vinci-client/src/main/java/com/linkedin/davinci/ingestion/IsolatedIngestionBackend.java

```diff
@@ -154,21 +154,24 @@ public void dropStoragePartitionGracefully(
       int timeoutInSeconds,
       boolean removeEmptyStorageEngine) {
     String topicName = storeConfig.getStoreVersionName();
-    mainIngestionMonitorService.cleanupTopicPartitionState(topicName, partition);
-    executeCommandWithRetry(
-        topicName,
-        partition,
-        REMOVE_PARTITION,
-        () -> mainIngestionRequestClient.unsubscribeTopicPartition(topicName, partition),
-        () -> {
-          super.dropStoragePartitionGracefully(storeConfig, partition, timeoutInSeconds, removeEmptyStorageEngine);
-          // Clean up the topic partition ingestion status.
-          mainIngestionRequestClient.resetTopicPartition(topicName, partition);
-        });
-    if (mainIngestionMonitorService.getTopicPartitionCount(topicName) == 0) {
+    executeCommandWithRetry(topicName, partition, REMOVE_PARTITION, () -> {
+      boolean result = getMainIngestionRequestClient().removeTopicPartition(topicName, partition);
+      // We will only clean up topic partition status if the remote execution is successful.
+      if (result) {
+        getMainIngestionMonitorService().cleanupTopicPartitionState(topicName, partition);
+        getMainIngestionRequestClient().resetTopicPartition(topicName, partition);
+      }
+      return result;
+    }, () -> {
+      removeTopicPartitionLocally(storeConfig, partition, timeoutInSeconds, removeEmptyStorageEngine);
+      // We will only clean up topic partition status if the local execution is successful.
+      getMainIngestionMonitorService().cleanupTopicPartitionState(topicName, partition);
+      getMainIngestionRequestClient().resetTopicPartition(topicName, partition);
+    });
```

Comment:
Nitpick: IDK if it is spotless playing games on you, but this change might be easier to review if we did not change the syntax (i.e. left the params one per line). In reality, less code changed here than it appears. Anyway, not a big deal... just mentioning it for the future.